vendor/github.com/openshift/library-go/pkg/network/dialer_linux.go

```diff
@@ -16,28 +15,18 @@ func dialerWithDefaultOptions() DialContext {
 	nd := &net.Dialer{
 		// TCP_USER_TIMEOUT does affect the behaviour of connect() which is controlled by this field so we set it to the same value
 		Timeout: 25 * time.Second,
+		// KeepAlive must to be set to a negative value to stop std library from applying the default values
+		// by doing so we ensure that the options we are interested in won't be overwritten
+		KeepAlive: time.Duration(-1),
```

Comment:
it gets overwritten here https://github.com/golang/go/blob/master/src/net/dial.go#L431